site/_posts/2020-11-15-how-to-securely-implement-oauth-rails.md

```diff
@@ -0,0 +1,300 @@
+---
+layout: blog-post
+title: How to securely implement OAuth in Ruby on Rails 
+description: A detailed overview of securely integrating an Ruby on Rails application with an OAuth provider using the OAuth Authorization Code Grant  
+author: Brett Guy 
+image: blogs/fusionauth-example-angular/oauth-angular-fusionauth.png
+category: blog
+tags: client-ruby
+excerpt_separator: "<!--more-->"
+---
+
+In this tutorial, we'll walk through setting up a basic Ruby on Rails app to securely authenticate with an OAuth2 server using the authorization code grant.
+
+<!--more-->
+
+Many Rails applications traditionally handle authentication, authorization, and user-management within the framework itself. There are many fine strategies for implementing auth whether it be homegrown or using a handy gem like [devise](https://github.com/heartcombo/devise). 
+With FusionAuth however, we are able to [separate our auth concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) from our application. Right away, we can scale our user base independent of our main application. As we build new applications or integrate with other platforms, 
```

Comment:
Also, why might I do this? My company has an existing idp and I need to integrate with it. Or I'm doing a startup and I want to offload all user management (functionality and security concerns) to FusionAuth.

Or ... insert your own reason. The benefit of saying a why up top is that you can then refer to it further down below to tie together the post.